### PR TITLE
feat(component): mobile menu support

### PR DESCRIPTION
**Pull request by CatsJuice**
close AF-1246

- Remove `<MenuIcon>` usage, add `prefixIcon`, `suffixIcon` for `<MenuItem>`
- Simplify `<MenuTrigger>`
- Render different `Menu` for desktop/mobile
- Use new design system for Menu
- Update Menu layout with `flex` and `gap`
- Add different `animation` for `<Modal>`
- bump `@toeverything/theme` ( some variables renamed)

### CODE REVIEW

**Comment by CatsJuice** [comment]
* **#7892** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7892?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7891** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7891?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CatsJuice and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by pengx17 — APPROVED** [review]
The abstraction looks good.
One minor issue looks like the backdrop flashes when closing.

**Comment by CatsJuice** [comment]
> The abstraction looks good.
> One minor issue looks like the backdrop flashes when closing.

Seems happen when open and close quickly, should be the issue of modal's view-transition animation.

File: packages/frontend/component/src/ui/menu/mobile/root.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  const onItemClick = useCallback((e: any) => {
+    e.preventDefault();
+    setOpen(prev => !prev);
+  }, []);
+
+  // dynamic height for slider
+  useEffect(() => {
+    if (!finalOpen) return;
+
+    const t = setTimeout(() => {
```
Comment (EYHN) [inline packages/frontend/component/src/ui/menu/mobile/root.tsx:87]: what about resize observer?

**Review by EYHN — APPROVED** [review]

**Pull request merged**